File: examples/tle/batch_propagation.py
# /// script
# dependencies = ["brahe"]
# ///
# FLAGS = [IGNORE]
"""
Batch propagation of multiple satellites.

Demonstrates:
- Getting ephemeris as propagators directly
- Propagating multiple satellites to same epoch
- Working with propagator collections
"""

import brahe as bh

if __name__ == "__main__":
    # Initialize EOP (required for propagators)
    eop = bh.StaticEOPProvider.from_zero()
    bh.set_global_eop_provider_from_static_provider(eop)

    # Get ephemeris and initialize propagators directly
    propagators = bh.datasets.celestrak.get_ephemeris_as_propagators(
        "gnss", step_size=60.0
    )

    print(f"Loaded {len(propagators)} GNSS propagators")

    # Propagate all satellites to same epoch
    epoch = bh.Epoch.from_datetime(2021, 1, 2, 12, 0, 0, 0, bh.TimeSystem.UTC)
    states = [prop.state(epoch) for prop in propagators]

    print(f"Propagated {len(states)} satellites to {epoch}")
    print(f"First satellite position: {states[0][:3] / 1000} km")

File: examples/tle/batch_propagation.rs
//! Batch propagation of multiple satellites.
//!
//! Demonstrates:
//! - Getting ephemeris as propagators directly
//! - Propagating multiple satellites to same epoch
//! - Working with propagator collections
//!
//! FLAGS = [IGNORE]

use brahe::datasets::celestrak;
use brahe::eop::*;
use brahe::orbits::traits::StateProvider;
use brahe::time::*;

fn main() {
    // Initialize EOP (required for propagators)
    let eop = StaticEOPProvider::from_zero();
    set_global_eop_provider(eop);

    // Get ephemeris and initialize propagators directly
    let propagators = celestrak::get_ephemeris_as_propagators("gnss", 60.0).unwrap();

    println!("Loaded {} GNSS propagators", propagators.len());

    // Propagate all satellites to same epoch
    let epoch = Epoch::from_datetime(2021, 1, 2, 12, 0, 0.0, 0.0, TimeSystem::UTC);
    let states: Vec<_> = propagators.iter().map(|prop| prop.state(epoch)).collect();

    println!("Propagated {} satellites to {}", states.len(), epoch);
    println!("First satellite position: [{:.3}, {:.3}, {:.3}] km",
        states[0][0] / 1000.0, states[0][1] / 1000.0, states[0][2] / 1000.0);
}
